### PR TITLE
fix(configuration): remove deprecated macros & add new ones

### PR DESCRIPTION
**Pull request by selfakiri**
## Description

Configuration : Remove deprecated macros & add new ones

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x
- [x] master

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by adr-mo — CHANGES_REQUESTED** [review]
Hi @selfakiri you have handled **the fresh installation of Centreon** but you are missing the upgrade.
On upgrade to the next minor / major version nothing will be changed without an upgrade script

**Comment by selfakiri** [comment]
> Hi @selfakiri you have handled **the fresh installation of Centreon** but you are missing the upgrade. On upgrade to the next minor / major version nothing will be changed without an upgrade script

Hello @adr-mo, I have a few questions regarding the upgrade process:
1. Should I handle the upgrade in this PR or in the backports?
2. Regarding update scripts, should I:
a) Add scripts for all minor versions (from zero to nine), or
b) Create one script per major version (only for 203.10.0 and 24.04.0)?

File: centreon/www/install/php/Update-24.09.0.php [hunk 1]
```diff
@@ -208,6 +208,32 @@
     );
 };
 
+$updateNagiosMacros =  function (CentreonDB $pearDB) use (&$errorMessage): void {
```
Comment (callapa) [inline centreon/www/install/php/Update-24.09.0.php:211]: Your script must be idempotent.

**Review by callapa — CHANGES_REQUESTED** [review]

File: centreon/www/install/php/Update-24.10.1.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                '$NOTIFICATIONCOMMENT$'
+            )
+        SQL
+    );
+
+    $errorMessage = 'Unable to insert new macros into nagios_macro table';
+    if (0 === (int) $statement->fetch(\PDO::FETCH_COLUMN)) {
+        $pearDB->executeQuery(
+        <<<'SQL'
+            INSERT IGNORE INTO `nagios_macro` (`macro_name`)
```
Comment (callapa) [inline centreon/www/install/php/Update-24.10.1.php:49]:
```suggestion
            INSERT INTO `nagios_macro` (`macro_name`)
```

**Review by callapa — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**